src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs

```diff
@@ -103,6 +111,15 @@ public void Close()
     /// <inheritdoc/>
     public bool HandlePartialDiscoveryComplete(IProxyDiscoveryManager proxyDiscoveryManager, long totalTests, IEnumerable<TestCase> lastChunk, bool isAborted)
     {
+        // Ensures that the total count of sources remains the same between each discovery
+        // completion of the same initial discovery request.
+        Debug.Assert(
```

Comment:
Maybe some formatting for the message could tell me what counts were reported and what was expected?